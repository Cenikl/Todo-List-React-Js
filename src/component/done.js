import React from "react";

export const Done = (props) => {
const{finished} = props;

    return(<div data-testid="done-1" className="done">
    <h1>Done</h1>
    <hr></hr>
    <ul>
      {finished.map(e => {
        return(
          <li key={e.title} className="listing">
            <p> <h4> Title : </h4> {e.title}</p>
          </li>
        )
      })}
    </ul>
  </div>)
}