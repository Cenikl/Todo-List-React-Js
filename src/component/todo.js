import React, { useState } from 'react';

export const Todo = (props) => {
  const {items,setItems,finished,setFinished,newTitle,setNewTitle} = props;

  const addItem = (event) => {
    if(event.key === 'Enter'){
    if(!newTitle){
      alert("Enter a todo title");
      return;
    }

    const item = {
      title: newTitle
    }; 
    setItems(oldList => [...oldList,item]);
    setNewTitle("");
  }
  }

  const finishWork = (title) => {
    const newFinished = items.filter(item => item.title === title);
    const newArray = items.filter(item => item.title !== title);
    setItems(newArray);
    const finishedItem = {
      title: newFinished[0].title
    }
    setFinished(oldList => [...oldList,finishedItem]);
  }

  
  

  return(
    <div data-testid="todo-1" className="todo">
        <h1>Todo List App</h1>
        <hr></hr>
        <div className="inputing">
        <input type={'text'} 
          placeholder="Put a Title here..." 
          value={newTitle}
          onChange= {e => setNewTitle(e.target.value)}
          onKeyDown={addItem}
          className="titre"/>
          <br/>
        </div>
        <ul>
          {items.map(item => {
            return(
              <li key={item.title} className="listing">
                <p> <h4> Title : </h4> {item.title}</p>
                <input type={'checkbox'} onClick={() => {finishWork(item.title)}}/>
              </li>
            )
          })}
        </ul>
      </div>
  )
}
