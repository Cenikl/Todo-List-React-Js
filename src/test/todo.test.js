import React from "react";
import { Todo } from "../component/todo";
import {render,screen,cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(()=>{
    cleanup();
})

it("renders without crashing",()=>{
    const items = [];
    const finished = [];
    const newTitle = "";
    render(<Todo items={items} finished={finished} newTitle={newTitle}/>);
    const todoElement = screen.getByTestId("todo-1");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("Todo List App");
})
it("renders one element",()=>{
    const items = [{title:"test"}]
    const finished = [];
    const newTitle = "";
    render(<Todo items={items} finished={finished} newTitle={newTitle}/>);
    const todoElement = screen.getByTestId("todo-1");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("test");

})
it("renders two element",()=>{
    const items = [{title:"test"},{title:"testing"}]
    const finished = [];
    const newTitle = "";
    render(<Todo items={items} finished={finished} newTitle={newTitle}/>);
    const todoElement = screen.getByTestId("todo-1");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("test");
    expect(todoElement).toHaveTextContent("testing");
})
it("macthes snapshot 1",()=>{
    const items = [];
    const tree = renderer.create(<Todo items={items}/>).toJSON();
    expect(tree).toMatchSnapshot();
})
it("macthes snapshot 2",()=>{
    const items = [{title:"test"},{title:"testing"}];
    const tree = renderer.create(<Todo items={items}/>).toJSON();
    expect(tree).toMatchSnapshot();
})