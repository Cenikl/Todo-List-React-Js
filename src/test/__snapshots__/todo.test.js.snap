// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macthes snapshot 1 1`] = `
<div
  className="todo"
  data-testid="todo-1"
>
  <h1>
    Todo List App
  </h1>
  <hr />
  <div
    className="inputing"
  >
    <input
      className="titre"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Put a Title here..."
      type="text"
    />
    <br />
  </div>
  <ul />
</div>
`;

exports[`macthes snapshot 2 1`] = `
<div
  className="todo"
  data-testid="todo-1"
>
  <h1>
    Todo List App
  </h1>
  <hr />
  <div
    className="inputing"
  >
    <input
      className="titre"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Put a Title here..."
      type="text"
    />
    <br />
  </div>
  <ul>
    <li
      className="listing"
    >
      <p>
         
        <h4>
           Title : 
        </h4>
         
        test
      </p>
      <input
        onClick={[Function]}
        type="checkbox"
      />
    </li>
    <li
      className="listing"
    >
      <p>
         
        <h4>
           Title : 
        </h4>
         
        testing
      </p>
      <input
        onClick={[Function]}
        type="checkbox"
      />
    </li>
  </ul>
</div>
`;
